crates/turborepo-lib/src/shim.rs

```diff
@@ -95,18 +101,30 @@ impl ShimArgs {
             Ok(ShimArgs {
                 cwd,
                 skip_infer,
+                force_update_check,
                 remaining_turbo_args,
                 forwarded_args,
             })
         }
     }
 
     // returns true if any flags result in pure json output to stdout
-    pub fn has_json_flags(&self) -> bool {
+    fn has_json_flags(&self) -> bool {
         self.remaining_turbo_args
             .iter()
             .any(|arg| TURBO_PURE_OUTPUT_ARGS.contains(&arg.as_str()))
     }
+
+    // returns true if any flags should bypass the update notifier
+    fn has_notifier_skip_flags(&self) -> bool {
+        self.remaining_turbo_args
+            .iter()
+            .any(|arg| TURBO_SKIP_NOTIFIER_ARGS.contains(&arg.as_str()))
+    }
+
+    pub fn should_skip_updater(&self) -> bool {
+        !self.force_update_check && (self.has_json_flags() || self.has_notifier_skip_flags())
```

Comment:
This isn't bad but it took me a sec to run through the possibilities. It might be nicer to have an explicit early exit for the `self.force_update_check` like:

```
if self.force_update_check {
  return false;
}
```